mantidproject/mantid
# Indirect - QENS Fitting - QENSFitSequential algorithm

This PR adds a more general algorithm for performing QENS sequential fitting - QENSFitSequential. This algorithm performs all common functionality found previously in each of the QENS sequential fitting algorithms (ConvolutionFitSequential, IqtFitSequential, MSDFit).

In addition, this algorithm has additional input properties (completely matching those of PlotPeakByLogValue) and is able to accept multiple datasets and discontinuous spectra ranges as input; not previously possible within the QENS sequential fitting algorithms.

This algorithm fully encompasses the functionality of the MSDFit algorithm -- and MSDFit's use in the QENS fitting interfaces has therefore been replaced. Similarly, this algorithm is used in the JumpFit interface.

**To test:**
1. Ensure the QENSFitSequential, ConvolutionFitSequential and IqtFitSequential unit tests pass.
2. Follow the test instructions for each of the following tabs in Interfaces > Indirect > Data Analysis:
    - MSDFit:
        1. In the 'Input' field, select the file found in the MSDFit folder (found in the test files).
        2. In the function browser, change the 'Fit Type' to 'Gaussian'.
        3. Click 'Run' and ensure the algorithm completes successfully.
    - IqtFit:
        1. In the 'Input' field, select the file found in the IqtFit folder (found in the test files).
        2. In the function browser, select 1 in the 'Exponential' spinner.
        3. Change 'StartX' to 0.16.
        4. Click 'Run' and ensure the algorithm completes successfully.
    - ConvFit:
        1. In the 'Sample' field, select the '\*\_red.nxs' file found in the ConvFit folder (found in the test files).
        2. In the 'Resolution' field select the '\*\_res.nxs' file found in the ConvFit folder (found in the test files).
        3. Select 'One Lorentzian' from the drop-down menu labelled 'Fit Type'.
        4. Check the check-box labelled 'Use Delta Function'.
        5. Check the check-box labelled 'Extract Members'.
        5. Click 'Run' and ensure the algorithm completes successfully and a \*\_Members workspace is created.
    - JumpFit:
        1. In the 'Input' field, select the file found in the JumpFit folder (found in the test files).
        2. In the function browser, change the 'Fit Type' to any type.
        3. Click 'Run' and ensure the algorithm completes successfully.



Fixes #22275
Refs #22303

**Test Files:**
[QENSFitTestFiles.zip](https://github.com/mantidproject/mantid/files/1971362/QENSFitTestFiles.zip)

**Release Notes** 
Changes have been added to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
